Source/DafnyCore/Resolver/HigherOrderHeapAllocationChecker.cs

```diff
@@ -0,0 +1,93 @@
+//-----------------------------------------------------------------------------
+//
+// Copyright by the contributors to the Dafny Project
+// SPDX-License-Identifier: MIT
+//
+//-----------------------------------------------------------------------------
+
+namespace Microsoft.Dafny;
+
+// This checker prevents the definition of non-terminating functions
+// by storing functions in memory (aka Landin's knots)
+// Thanks to frame information, we need not reject all assignments of
+// functions to memory, but only the ones that are know to have a 
+// read frame.
```

Comment:
To make these into doc comments, change `//` into `///` and sandwich these lines between

``` c#
/// < summary>
...
/// </summary>
```